### PR TITLE
Adds documentation on releasing

### LINKED ISSUE #592: Document release process
/kind documentation

Document the release process. Probably need to sync with @detiber to get all the details.

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #592 

**Special notes for your reviewer**:
This is a rough draft and any updates to the documented process would be greatly appreciated.

```release-note
NONE
```

### CODE REVIEW

File: docs/releasing.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Release process
+
+## Manual steps
+
+### Expected artifacts
+
+1. A container image containing the aws-provider manager binary
+2. A release tarball containing the manifest-templates and a script to generate
```
Comment (detiber) [inline docs/releasing.md]: This is missing `clusterctl` and `clusterawsadm` binaries for Linux and Mac

File: docs/releasing.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   the actual manifests
+3. Release notes
```
Comment (detiber) [inline docs/releasing.md]: Should we also list the actual github release here? We *could* also list the git tag here as well, but that can be implicitly implied from the github release.

File: docs/releasing.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+The container image will live in the registry `gcr.io/cluster-api-provider-aws`
+under the image name `cluster-api-aws-controller:<tag>` where `<tag>` is
+replaced by the version being released.
+
+#### Manifests
+
+Manifests must be generated by hand.
+
+Running `make release-artifacts` will create a tarball that you can attach to
```
Comment (detiber) [inline docs/releasing.md]: this also generates the `clusterctl` and `clusterawsadm` binaries.

File: docs/releasing.md (hunk 2 above)
Comment (chuckha) [reply docs/releasing.md]: The tag imo is less important since the repo isn't designed to be imported from another library

**Pull request merged**